daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -462,10 +464,11 @@ private[speedy] case class PartialTransaction(
           case Some(KeyActive(_)) => KeyConflict.Duplicate
           case Some(KeyInactive) | None => KeyConflict.None
         }
-        val globalKeyInputs = keys.get(ck).orElse(ptx.globalKeyInputs.get(ck)) match {
-          case None => ptx.globalKeyInputs.updated(ck, KeyInactive)
-          case Some(_) => ptx.globalKeyInputs
-        }
+        val globalKeyInputs =
+          if (ptx.globalKeyInputs.isDefinedAt(ck))
+            ptx.globalKeyInputs.updated(ck, KeyInactive)
```

Comment:
isn’t this the wrong way around? We want to update if it's _not_ defined.

Reply:
Good catch.